File: gradle/src/main/kotlin/HelloWorld.kt
// Hello World in Gradle !

package hello

import org.apache.commons.lang3.time.DateFormatUtils

fun printGreeting(args: Array<String>) {
    if (args.isNotEmpty()) {
        val name = args.drop(1).joinToString(" ")
        println("Hi $name, greetings from Gradle !")
    } else {
        println("Hello World from Gradle !")
    }
}

fun printDate() {
    val currentDate = System.currentTimeMillis()
    val format = "MMMM dd, yyyy HH:mm:ss"
    val formattedDate = DateFormatUtils.format(currentDate, format)
    println("Today is $formattedDate")
}

fun main(args: Array<String>) {
    printGreeting(args)
    printDate()
}

File: dotnet/src/HelloWorld/HelloWorld.cs
namespace Hello.World;

public class HelloWorld
{
    public static string PrintGreeting(string[] args)
    {
        if (args.Length > 0)
        {
            string name = string.Join(" ", args);
            return $"Hi {name}, greetings from Dotnet !";
        }
        else
        {
            return "Hello World from Dotnet !";
        }
    }

    public static string PrintDate()
    {
        DateTime currentDate = DateTime.Now;
        string layout = "MMMM d, yyyy 'at' hh:mm tt";
        string formattedDate = currentDate.ToString(layout);
        return $"Today is {formattedDate}";
    }
}
